### DISCUSSION TITLE
wanted help in running a pretrained keras model

### DISCUSSION THREAD

**Discussion by ft10101** [community/community#78498] [Programming Help]
https://github.com/ft10101/Projects/blob/main/LSTM_model_(shakespare)(1).py

assume that the same model is saved in a .keras file after training and how to use this now ?

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @ft10101 ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by faridnec** [comment]
Hey!
After training, when you save your model using something like model.save("shakespeare_lstm_model.keras"), it stores the architecture, weights, and optimizer state.

Now, when you want to use this saved model for predictions or further training, it's pretty straightforward. You load it using loaded_model = keras.models.load_model("shakespeare_lstm_model.keras").

Once loaded, you can throw new text at it for predictions like this (just for an instance)
```python
text = "To be or not to be"
processed_text = preprocess_text(text)  #  need to define preprocess_text based on how you preprocessed during training
prediction = loaded_model.predict(processed_text)
```

Or if you want to continue training:
```python
loaded_model.fit(new_data, new_labels, epochs=5)  # Adjust the epochs and data accordingly
```

Just make sure your text preprocessing during predictions matches what you did during training.
